### PR TITLE
Scraper 12x Performance Increase with Promises

### PR DESCRIPTION
**Pull request by canarado**
make sure i didnt fuck anything up 👍

### CODE REVIEW

**Comment by canarado** [comment]
Once I've thought it over, 12x perf increase is a bit subjective. It's more like <insert your cpu thread count> x perf increase. Just fyi

File: api/routes/index.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 const express = require('express');
 const axios = require('axios');
 const router = express.Router();
-const scraper = require(`${__dirname}/../../scraper/scraper`);
```
Comment (IllusionMan1212) [inline api/routes/index.js:4]: I'd rather the `__dirname` is kept

File: api/routes/index.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
+const { scrapeCheese, getCheeseOfDay, getCheeseInfo } = require('../../scraper/scraper');
 const { JSDOM } = require('jsdom');
 
-const Cheese = require(`${__dirname}/../models/cheese`);
```
Comment (IllusionMan1212) [inline api/routes/index.js:7]: ditto

File: api/routes/index.js [hunk 3]
```diff
@@ -120,53 +120,74 @@ router.get('/cheese/vegetarian', (req, res) => {
...
-                });
-            }
+            });
         }
-    }
-    res.status(200).json({ failed: false, status: 200, cheeses: all_cheeses });
+
+        res.status(200).json({ failed: false, status: 200, cheeses: _res });
+    })
+    // let pages = 1;
```
Comment (IllusionMan1212) [inline api/routes/index.js]: i dont like commented code, but it's not a big deal

File: scraper/scraper.js [hunk 4]
```diff
@@ -2,210 +2,348 @@ const axios = require("axios");
 const { JSDOM } = require("jsdom");
 
 let country_codes_map = {
-    AF: "afghanistan",
```
Comment (IllusionMan1212) [inline scraper/scraper.js]: 4 spaces plz :pepehands:

File: scraper/scraper.js [hunk 5]
```diff
@@ -2,210 +2,348 @@ const axios = require("axios");
...
+  return axios
+    .get(link)
+    .then((response) => {
+      const dom = new JSDOM(response.data);
+      const { document } = dom.window;
+      const attrib_arr = Array.from(
+        document
+          .getElementsByClassName("summary-points")[0]
+          .getElementsByTagName("p")
+      );
```
Comment (IllusionMan1212) [inline scraper/scraper.js]: big nitpick here but why is it broken down into 4 lines

File: scraper/scraper.js [hunk 6]
```diff
@@ -2,210 +2,348 @@ const axios = require("axios");
...
+        } else if (attrib.textContent.includes("Country of origin")) {
+          attribs.countries = Array.from(attrib.querySelectorAll("a")).map(
+            (elem) => {
+              countries.push(
+                Object.keys(country_codes_map).find(
+                  (key) =>
+                    country_codes_map[key] == elem.textContent.toLowerCase().trim()
+                )
+              );
+              return elem.textContent.trim();
```
Comment (IllusionMan1212) [inline scraper/scraper.js]: same here

File: scraper/scraper.js [hunk 7]
```diff
@@ -2,210 +2,348 @@ const axios = require("axios");
...
+          );
+        } else if (attrib.textContent.includes("Region")) {
+          attribs.region = attrib.textContent.split(":")[1].trim();
+        } else if (attrib.textContent.includes("Family")) {
+          attribs.family = attrib.textContent.split(":")[1].trim();
+        } else if (attrib.textContent.includes("Type")) {
+          attribs.types = attrib.textContent
+            .split(":")[1]
+            .split(",")
+            .map((elem) => elem.trim());
```
Comment (IllusionMan1212) [inline scraper/scraper.js]: these make sense imo so no need to change them

File: scraper/scraper.js [hunk 8]
```diff
@@ -2,210 +2,348 @@ const axios = require("axios");
...
+
+    Promise.all(a).then(a => {
+      let merged = [].concat.apply([], a);
+      resolve(merged)
+    })
+  })
 }
 
 async function getCheeseOfDay() {
-    return axios.get("https://cheese.com")
```
Comment (IllusionMan1212) [inline scraper/scraper.js:188]: spaces

**Review by IllusionMan1212 — CHANGES_REQUESTED** [review]
i got lots of complaints lmao

File: scraper/scraper.js (hunk 5 above)
Comment (canarado) [reply scraper/scraper.js]: That's right, you like messy hard to read code, I forgot. Smh, ill fix tomorrow

**Pull request merged**